apisix/plugins/fault-injection.lua

```diff
@@ -28,13 +30,15 @@ local schema = {
             properties = {
                 http_status = {type = "integer", minimum = 200},
                 body = {type = "string", minLength = 0},
+                percentage = {type = "integer", minimum = 0, maximum = 100}
```

Comment:
I am not sure if this way is power enough.

I am thinking, how about support `vars` for this feature?

`vars`, here is an example: `["random_100", ">", 70]`.

then we can support to use any request header or query string as condition, eg: `["arg_age", ">", 18]`

Reply:
Actually the percentage is inspired from Istio. I think the purpose of fault injection is to observe whether the whole request link is still normal when some parts of them don't work so perfectly,which may need the randomness to mimic the production, instead of the deliberate order from administrator.

Reply:
thx for your explain